tools/coverage.sh

```diff
@@ -4,22 +4,21 @@ PROJECT_DIR=$(git rev-parse --show-toplevel)
 MAX_NODE_SIZE=$1
 if [ -z ${MAX_NODE_SIZE} ]; then MAX_NODE_SIZE=1000; fi
 cd ${PROJECT_DIR}
-rm -rf html
 mkdir -p build.cov
 cd build.cov/
-echo cmake -G Ninja -D REALM_COVERAGE=ON -D CMAKE_CXX_FLAGS=-g -DREALM_MAX_BPNODE_SIZE=${MAX_NODE_SIZE} ..
-cmake -G Ninja -D REALM_COVERAGE=ON -D CMAKE_CXX_FLAGS=-g -DREALM_MAX_BPNODE_SIZE=${MAX_NODE_SIZE} ..
+if [ ! -f CMakeCache.txt ]; then
+  echo cmake -G Ninja -D REALM_COVERAGE=ON -D CMAKE_CXX_FLAGS=-g -DREALM_MAX_BPNODE_SIZE=${MAX_NODE_SIZE} ..
+  cmake -G Ninja -D REALM_COVERAGE=ON -D CMAKE_CXX_FLAGS=-g -DREALM_MAX_BPNODE_SIZE=${MAX_NODE_SIZE} ..
+fi
 ninja
-cd ${PROJECT_DIR}
+
 if [ ! -f coverage-base.info ]; then
-  lcov --no-external --capture --initial --directory . --output-file ./coverage-base.info
+  lcov --capture --initial --directory src --output-file ./coverage-base.info
 fi
-cd build.cov
 ctest 
-cd ${PROJECT_DIR}
-lcov --no-external --directory . --capture --output-file ./coverage-test.info
+lcov --directory src --capture --output-file ./coverage-test.info
 lcov --add-tracefile coverage-base.info --add-tracefile coverage-test.info --output-file coverage-total.info
-lcov --remove coverage-total.info "/usr/*" "*/test/*" "*/external/*" "*/generated/*" "*query_flex*" --output-file coverage-filtered.info
-genhtml coverage-filtered.info --output-directory html
-firefox html/index.html > /dev/null 2>&1 &
+lcov --remove coverage-total.info "/usr/*" "*/external/*" "*/generated/*" "*query_flex*" --output-file coverage-filtered.info
+genhtml coverage-filtered.info
+firefox index.html > /dev/null 2>&1 &
```

Comment:
I can see this was here previously, but we could use a more generic version based on the user's default browser (not blocking):
```suggestion
open index.html
```

Reply:
`open` does not work on Linux